modules/caddyhttp/fileserver/browsetplcontext.go

```diff
@@ -82,6 +82,16 @@ func (fsrv *FileServer) directoryListing(ctx context.Context, entries []fs.DirEn
 
 		u := url.URL{Path: "./" + name} // prepend with "./" to fix paths with ':' in the name
 
+		// get extension(s)
+		var ext, fullExt string
+		firstDot, lastDot := strings.Index(name, "."), strings.LastIndex(name, ".")
+		if firstDot >= 0 {
+			fullExt = name[firstDot:]
+		}
+		if lastDot >= 0 {
+			ext = name[lastDot:]
+		}
```

Comment:
Why not?

```suggestion
		// get extension(s)
		ext := filepath.Ext(name)
```

Reply:
Because `.gz` and `.tar.gz` are different, and if we want to show a different icon for those, we need to know what the "full" extension is. (We don't currently do this, but I want to as soon as an icon becomes available for that.)

Reply:
Gotcha. Fair enough. I can see this being problematic if some file has the name `agreement.document.v1.docx`, but it isn't a breaker.

Reply:
Ok, now much better :)